skimage/transform/_warps.py

```diff
@@ -778,6 +760,32 @@ def warp(image, inverse_map, map_args={}, output_shape=None, order=1,
 
         if matrix is not None:
             matrix = matrix.astype(np.double)
+
+            if resize:
+                rows, cols = input_shape[:2]
```

Comment:
I'm not sure what is 3-D images. Its something like arrays which have shape `(512, 512, 512)`?
If so, how can I detect the input image is 3-D or multichannel? RGB image is `(512, 512, 3)` but its not 3-D image, right?